### ISSUE TITLE
Editing profile does not work

### ISSUE THREAD

**Issue by eloquence** [internetarchive/openlibrary#613]
I tried to edit my profile via https://openlibrary.org/people/xirzon?m=edit , but I get redirected to a 405 error page with only the text "None". Here are the request headers:

````
Accept:text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Encoding:gzip, deflate, br
Accept-Language:en-US,en;q=0.9,de;q=0.8
Cache-Control:no-cache
Connection:keep-alive
Content-Length:149
Content-Type:application/x-www-form-urlencoded
Cookie:session=(redacted)
Host:openlibrary.org
Origin:https://openlibrary.org
Pragma:no-cache
Referer:https://openlibrary.org/people/xirzon?m=edit
Upgrade-Insecure-Requests:1
User-Agent:Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/62.0.3202.62 Chrome/62.0.3202.62 Safari/537.36
````

And this is the form payload:

````
type.key:/type/user
displayname:xirzon
type.key:/type/user
website#0:
description:test
_comment:
_save:
has_fulltext:true
has_fulltext:true
````

**Comment by cdrini** [comment]
I experience the same issue on my DriniBot account, but not on my main account. I notice that your account was created recently (as was DriniBot); perhaps this related to when account was registered?

**Comment by cdrini** [comment]
I think that error is coming from here: https://github.com/internetarchive/infogami/blob/a1430609d73316eaebeba0f723e17a82a1a71b8d/infogami/plugins/api/code.py#L33

**Comment by mekarpeles** [comment]
Thank you! Will look into this; just for transparency, flagging as low priority (so if someone wants to get to it before me, happy to code review)

**Comment by mekarpeles** [comment]
Tagging @dcapillae, @JeffKaplan, @kellyhogan, @tfmorris, and @Num170R on this issue because we were previously following duplicates of this issue which were closed

**Comment by LeadSongDog** [comment]
Maybe not such a low priority?

**Comment by Num170R** [comment]
I tend to think that this is chromium specific. At least I didn't manage to reproduce using different browsers. Can anyone confirm?

**Comment by dcapillae** [comment]
@Num170R I think it's not a specific Chromium problem. I use Firefox and I have the same problem.

**Comment by LeadSongDog** [comment]
Is this confirmed to have been while logged in?

**Comment by dcapillae** [comment]
I can edit any page of a book, but it is not possible to change the information on my own profile page.

**Comment by Num170R** [comment]
@dcapillae which version of firefox are you using (version+os)?

**Comment by Num170R** [comment]
thanks for replying @dcapillae . probably that's why I couldn't replicate, regarding that you are using a fairly old firefox version. could you please also test with a recent firefox version (>=60) from `https://www.mozilla.org/en-US/firefox/` for reference?

**Comment by Num170R** [comment]
The latest update (Chrome/66) also fixed this issue in Chromium.
It seems as if the POST request was getting modified resulting to 405:

1. POST /people/openlibrary7449?m=edit&has_fulltext=true
2. POST /people/openlibrary7449?has_fulltext=true

Modifying on the fly the request in (2) to match (1) could successfully resolve the issue.

**Comment by mekarpeles** [comment]
@Num170R has totally solved this issue, a big 🙇 🎉  to him!

The problem is, our OpenLibrary `header` section search eagerly adds `has_fulltext=true` into the url and is presumably clobbering some get parameters on other pages.

I'm re-assigning this to myself since I cause the issue, and will hopefully have a solution this weekend.

**Comment by Num170R** [comment]
@mekarpeles I need to test a few more cases and I'll update soon

**Comment by mekarpeles** [comment]
open library profile page cannot be editted

@JeffKaplan reported a similar issue; re-opening for investigation:

after making change to `*?m=edit` page the submit returns `None`.
e.g.:

`https://openlibrary.org/people/riadess?m=edit`

**Comment by jdlrobson** [comment]
This looks fixed to me. I can't reproduce this anymore. The JS cleanup that has occurred may have rectified this...

**Comment by jdlrobson** [comment]
Looks like this was fixed by 74fb0785848c242625ddd9d7ebe8b31a481d8905
